### PR TITLE
esmodules: Update lib/route

### PR DESCRIPTION
**Pull request by dmsnell**
@see: https://github.com/Automattic/wp-calypso/milestone/224

Previously we were exporting a default object of multiple methods and
importing those through the non-spec-compliant Babel destructuring
import statements.

This patch makes those methods proper named exports in the work of
turning off CommonJS compilation.

### CODE REVIEW

File: client/my-sites/stats/controller.jsx [hunk 1]
```diff
@@ -111,30 +109,26 @@ export default {
...
 		const FollowList = require( 'lib/follow-list' );
-		let siteId = context.params.site_id;
-		const basePath = route.sectionify( context.path );
+		const basePath = sectionify( context.path );
 		const followList = new FollowList();
 
 		// FIXME: Auto-converted from the Flux setTitle action. Please use <DocumentHead> instead.
 		context.store.dispatch( setTitle( i18n.translate( 'Stats', { textOnly: true } ) ) );
 
-		const site = getSite( context.store.getState(), siteId );
```
Comment (samouri) [inline client/my-sites/stats/controller.jsx]: why is this change here

Comment (dmsnell) [reply client/my-sites/stats/controller.jsx]: I believe that it's because `siteId` wasn't being used, thus `site` wasn't needed

File: client/jetpack-connect/test/__snapshots__/main.js.snap [hunk 2]
```diff
@@ -20,12 +20,6 @@ Array [
 ]
 `;
 
-exports[`JetpackConnectMain goToPluginActivation should fire redirect 1`] = `
```
Comment (samouri) [inline client/jetpack-connect/test/__snapshots__/main.js.snap]: why is this file changing at all?

Comment (samouri) [reply client/jetpack-connect/test/__snapshots__/main.js.snap]: oh, they are obsolete snapshots.  lets fix that separately in a different PR.

**Comment by samouri** [comment]
i squashed, rebased, and removed some more unnecessary changes

File: client/jetpack-connect/test/main.js [hunk 3]
```diff
@@ -63,14 +65,18 @@ describe( 'JetpackConnectMain', () => {
...
+			addQueryArgs.mockReset();
 			externalRedirect.mockReset();
 		} );
 
 		test( 'should fire redirect', () => {
 			component.goToPluginActivation( 'example.com' );
 
 			expect( externalRedirect ).toHaveBeenCalledTimes( 1 );
-			expect( externalRedirect.mock.calls[ 0 ] ).toMatchSnapshot();
+			expect( addQueryArgs ).toHaveBeenCalledWith(
```
Comment (samouri) [inline client/jetpack-connect/test/main.js]: why is this here?

**Comment by samouri** [comment]
this PR seems fine from some minor smoke-testing locally.
If canaries and circleci are happy, then lets 🚢
